src/core/net/ip6_address.hpp

```diff
@@ -96,6 +97,53 @@ class InterfaceIdentifier : public otIp6InterfaceIdentifier,
      */
     bool IsReserved(void) const;
 
+    /**
+     * This method generates and sets the Interface Identifier to a crypto-secure random byte sequence.
+     *
+     */
+    void GenerateRandom(void);
+
+    /**
+     * This method gets the Interface Identifier as a pointer to a byte array.
+     *
+     * @returns A pointer to a byte array (of size `kSize`) containing the Interface Identifier.
+     *
+     */
+    const uint8_t *GetAsByteArray(void) const { return mFields.m8; }
+
+    /**
+     * This method sets the Interface Identifier from a given byte array.
+     *
+     * @param[in] aBuffer    Pointer to an array containing the Interface Identifier. `kSize` bytes from the buffer
+     *                       are copied to form the Interface Identifier.
+     *
+     */
+    void SetFromByteArray(const uint8_t *aBuffer);
+
+    /**
+     * This method sets the Interface Identifier from a given IEEE 802.15.4 Extended Address.
+     *
+     * @param[in] aExtAddress  An Extended Address.
+     *
+     */
+    void SetFromExtAddress(const Mac::ExtAddress &aExtAddress);
```

Comment:
nit: rename to `FromExtAddress` so that it's symmetric with `ToExtAddress`?

```suggestion
    void FromExtAddress(const Mac::ExtAddress &aExtAddress);
```

Reply:
Even though I think we don't explicitly mention it in our style guide, in terms of method names, I like that we start the name with a "verb/action" (I feel it reads better from the caller side - i.e., it's clear what action we are performing on the object). 

The only exception is `To{SomeType}()`, even then in some cases I actually prefer using `ConvertTo{SomeType}()` name. 

Here is my suggestion for this:
- We use `To{SomeType}()`name in cases the method converts and then returns the given type (like `ToString()` which returns a `String` instance).
- We use `ConvertTo{SomeType}(SomeType &)` method name in cases where the method convertsto a new type and outputs the converted value in a passed-in reference parameter.
- i.e.,
```
class Type
{
    OtherType ToOtherType(void) const;                    // First option (preferred)     
    void ConvertToOtherType(OtherType &aOtherType) const; // Second option (can return `otError` if conversion can fail)
    
    void ToOtherType(OtherType &aOtherType) const;        // Third option (not preferred).
};
```

I feel statement like 
```
    macAddr = meshLocalIid.ToExtAddress();
```
is more readable/clear compared to 
``` 
    meshLocalIid.ToExtAddress(macAddr);
```

With OT moving to C++11 and almost all toolchains supporting "return value optimization" and ["copy elision"](https://en.cppreference.com/w/cpp/language/copy_elision), the first option should not add any copying overhead and should be as efficient as the second option.

Thoughts? @jwhui?

Reply:
I would also prefer `OtherType ToOtherType(void) const;` if there is no performance cost.

Reply:
Agree. Use `CovertTo()` if the method needs to return error for any reason.